app/Main.hs

```diff
@@ -7,12 +7,16 @@ import Web.Spock.Config
 import Control.Monad.IO.Class (liftIO)
 import Data.Yaml.Config (loadYamlSettings, useEnv)
 import Data.Aeson (parseJSON, FromJSON, withObject, (.:))
+import Database.Persist.MySQL (MySQLConf)
 
-data AppSettings = AppSettings { appPort :: Int }
+data AppSettings = AppSettings { appPort :: Int
+                               , appDatabaseConf :: MySQLConf
+                               }
 
 instance FromJSON AppSettings where
   parseJSON = withObject "AppSettings" $ \o -> do
     appPort <- o .: "port"
+    appDatabaseConf <- o .: "database"
     return AppSettings{..}
```

Comment:
(question) would that be possible to reverse the order here and drop `do` syntax similar to what they do in [Argo](https://github.com/thoughtbot/Argo):
```
curry(User.init)
      <^> json <| "id"
      <*> json <| "name"
```